backend/plonk/bls12-377/setup.go

```diff
@@ -173,9 +146,13 @@ func (pk *ProvingKey) VerifyingKey() interface{} {
 	return pk.Vk
 }
 
-// BuildTrace fills the constant columns ql, qr, qm, qo, qk from the sparser1cs.
-// Size is the size of the system that is nb_constraints+nb_public_variables
-func BuildTrace(spr *cs.SparseR1CS, pt *Trace) {
+// NewTrace returns a new Trace object from the constraint system.
+// It fills the constant columns ql, qr, qm, qo, qk, and qcp with the
+// coefficients of the constraints.
+// Size is the size of the system that is next power of 2 (nb_constraints+nb_public_variables)
+// The permutation is also computed and stored in the Trace.
+func NewTrace(spr *cs.SparseR1CS, domain *fft.Domain) *Trace {
```

Comment:
Normally `domain.Cardinality` should match `size := ecc.NextPowerOfTwo(sizeSystem)`, don't know if we should add this sanity check...